app/connectors/GetCalculationListConnector.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.AppConfig
+import connectors.httpParsers.GetCalculationListHttpParser._
+import javax.inject.Inject
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GetCalculationListConnector @Inject()(http: HttpClient, val appConfig: AppConfig)(implicit ec: ExecutionContext) extends DesConnector {
+
+  def calcList(nino: String, taxYear: String, optionalTaxYear: Boolean)(implicit hc: HeaderCarrier): Future[GetCalculationListResponse] = {
```

Comment:
if taxYear is optional would be better to make taxYear an option[String] rather than a boolean alongside it

Reply:
@david-walters-HMRC any chance this could be looked into in a subsequent PR?